### PR TITLE
feat(write_luworkers): set the write LUWorkers using istgtcontrol

### PR DESCRIPTION
**Pull request by vishnuitta**
This PR is to limit the luworkers that can do simultaneous writes in replication module.
Command to change the setting: `istgtcontrol -t vol1 SET 16 <write_luworkers>`

Fixed a memory corruption in replication_test binary with read IOs during degraded mode.
Signed-off-by: Vishnu Itta <vitta@mayadata.io>

### CODE REVIEW

File: src/istgt_lu_ctl.c [hunk 1]
```diff
@@ -731,14 +752,18 @@ istgt_uctl_cmd_set(UCTL_Ptr uctl)
...
 			istgt_uctl_snprintf(uctl, "ERR no target\n");
 			goto error_return;
-		}
+		} else
+			ISTGT_NOTICELOG("found vol %s to do SET\n", iqn);
 		spec = lu->lun[0].spec;
 	}
 
 	switch (setopt) {
+#if 0
```
Comment (pawanpraka1) [inline src/istgt_lu_ctl.c:766]: it's better to use #if !defined(REPLICATION)

File: src/replication.c [hunk 2]
```diff
@@ -2812,6 +2822,13 @@ replicate(ISTGT_LU_DISK *spec, ISTGT_LU_CMD_Ptr cmd, uint64_t offset, uint64_t n
 
 	io_queue_time[cmd->luworkerindx].tv_sec =
 	    io_queue_time[cmd->luworkerindx].tv_nsec = 0;
+end:
+	if (cmd_write) {
+		pthread_mutex_lock(&spec->write_throttle_mtx);
+		spec->inflight_writes--;
+		pthread_cond_signal(&spec->write_throttle_cv);
```
Comment (pawanpraka1) [inline src/replication.c:2829]: only signal when there is someone waiting on it.

File: src/istgt_lu_ctl.c [hunk 3]
```diff
@@ -689,24 +689,45 @@ istgt_uctl_cmd_max_io_wait(UCTL_Ptr uctl)
...
 
+static int
+istgt_uctl_set_write_luworkers(ISTGT_LU_DISK *spec, int val)
+{
+	if (val < 0)
+		return -1;
+
+	if (spec != NULL) {
+		pthread_mutex_lock(&spec->write_throttle_mtx);
+		spec->write_luworkers = val;
```
Comment (pawanpraka1) [inline src/istgt_lu_ctl.c:700]: probably a check  to verify write_luworkers is < luworkers.

**Review by pawanpraka1 — APPROVED** [review]
looks good.

**Comment by vishnuitta** [comment]
reworking on this to avoid unnecessary signaling

**Comment by vishnuitta** [comment]
closing this to raise different PR with different approach

**Pull request closed without merging**